Classes/Emogrifier.php

```diff
@@ -323,6 +334,81 @@ protected function process(\DOMDocument $xmlDocument)
+                // Always override HTML attributes. CSS takes precedence.
+                $node->setAttribute('bgcolor', $styleValue);
+            } elseif ($styleName == 'background') {
+                // Parse out the color, if any
+                $styles = explode(' ', $styleValue);
+                $first = $styles[0];
+                if (!is_numeric(substr($first, 0, 1)) && substr($first, 0, 3) != 'url') {
+                    // This is not a position or image, assume it's a color
+                    $node->setAttribute('bgcolor', $first);
+                }
+            } elseif ($styleName == 'width') {
+                $width = preg_replace('/[^0-9.%]/', '', $styleValue);
+                $node->setAttribute('width', $width);
+            } elseif ($styleName == 'height') {
+                $height = preg_replace('/[^0-9.%]/', '', $styleValue);
+                $node->setAttribute('height', $height);
+            } elseif ($styleName == 'margin') {
+                $margins = explode(' ', $styleValue);
+                if (count($margins) == 1) {
+                    if ($margin[0] == 'auto') {
+                        $node->setAttribute('align', 'center');
+                    }
+                } elseif (count($margins) == 2) {
+                    if ($margin[1] == 'auto') {
+                        $node->setAttribute('align', 'center');
+                    }
+                } elseif (count($margins) == 3) {
+                    if ($margin[1] == 'auto') {
+                        $node->setAttribute('align', 'center');
+                    }
+                } else {
+                    if ($margin[1] == 'auto' && $margins[3] == 'auto') {
+                        $node->setAttribute('align', 'center');
+                    }
+                }
+            } elseif ($styleName == 'text-align' &&
+                        ($node->nodeName == 'p' || $node->nodeName == 'td' || $node->nodeName == 'div')) {
+                if ($styleValue == 'left' || $styleValue == 'right' ||
+                        $styleValue == 'center' || $styleValue == 'justify') {
+                    // Only set values that are valid for the align attribute
+                    $node->setAttribute('align', $styleValue);
+                }
+            } elseif ($styleName == 'float' && ($node->nodeName == 'table' || $node->nodeName == 'img')) {
+                if ($styleValue == 'left' || $styleValue == 'right') {
+                    // Only set values that are valid for the align attribute
+                    $node->setAttribute('align', $styleValue);
+                }
+            } elseif ($styleName == 'border') {
+                if ($styleValue == 'none' || $styleValue == '0') {
+                    $node->setAttribute('border', '0');
+                }
+            } elseif ($styleName == 'border-spacing' && $node->nodeName == 'table') {
+                $node->setAttribute('cellspacing', $styleValue);
+            }
```

Comment:
Hello Oliver,

Do you plan to pull this feature on the main branch soon? If not, I'd 
like to reorganize my branches a little bit.

Unfortunately, after forking I just started developing on my master 
branch and everything I do on master will go into this pull request. So 
I've been holding off on doing more changes until it was merged. If you 
need more time, I'll close the current pull request and create it again 
on a feature branch of my fork so I can continue on my master branch.

Regards,
Sander Kruger.

------------------------------------------------------------------------

*From:* Oliver Klee
*Sent:* Monday, September 28, 2015 9:45AM
*To:* Jjriv/emogrifier
*Cc:* Sander Kruger
*Subject:* Re: [emogrifier] [FEATURE] Add CSS to HTML attribute mapping 
(#232)

In Classes/Emogrifier.php 
<https://github.com/jjriv/emogrifier/pull/232#discussion_r40524553>:

> +                        $styleValue == 'center' || $styleValue == 'justify') {
> +                    // Only set values that are valid for the align attribute
> +                    $node->setAttribute('align', $styleValue);
> +                }
> +            } elseif ($styleName == 'float' && ($node->nodeName == 'table' || $node->nodeName == 'img')) {
> +                if ($styleValue == 'left' || $styleValue == 'right') {
> +                    // Only set values that are valid for the align attribute
> +                    $node->setAttribute('align', $styleValue);
> +                }
> +            } elseif ($styleName == 'border') {
> +                if ($styleValue == 'none' || $styleValue == '0') {
> +                    $node->setAttribute('border', '0');
> +                }
> +            } elseif ($styleName == 'border-spacing' && $node->nodeName == 'table') {
> +                $node->setAttribute('cellspacing', $styleValue);
> +            }

Generally, I find this method very long. I propose using either a switch 
or a look-up table instead (or a combination).

—
Reply to this email directly or view it on GitHub 
<https://github.com/jjriv/emogrifier/pull/232/files#r40524553>.